src/features/projectsV2/ProjectsMap/index.tsx

```diff
@@ -26,13 +31,28 @@ export type ProjectsMapProps = ProjectsMapMobileProps | ProjectsMapDesktopProps;
 function ProjectsMap(props: ProjectsMapProps) {
   const mapRef: MutableRefObject<null> = useRef(null);
   const { viewState, setViewState, mapState, mapOptions } = useProjectsMap();
+  const [isOnSampleMarker, setIsOnSampleMarker] = useState(false);
   const {
     plantLocations,
     setHoveredPlantLocation,
     setSelectedPlantLocation,
     setSelectedSite,
+    setSelectedSamplePlantLocation,
   } = useProjects();
   const { projects, singleProject, selectedPlantLocation } = useProjects();
+  const router = useRouter();
+  const locale = useLocale();
+  const updateSiteAndUrl = (
+    locale: string,
+    projectSlug: string,
+    siteIndex: number
+  ) => {
+    if (singleProject?.sites?.length === 0) return;
+    setSelectedSite(siteIndex);
+    const siteId = singleProject?.sites?.[siteIndex].properties.id;
+    if (siteId) updateUrlWithSiteId(locale, projectSlug, siteId, router);
+  };
+
```

Comment:
Why was this added here, when we already have similar code in the ProjectsProvider?

My concern is that we are keeping the logic that updates the url in too many places, which could cause bugs down the line.